gr-fec/lib/cc_decoder_impl.cc

```diff
@@ -64,7 +64,7 @@ cc_decoder_impl::cc_decoder_impl(int frame_size,
       d_start_state_nonchaining(start_state),
       d_end_state_nonchaining(end_state)
 {
-    if (k != 7 || rate != 2) {
+    if (k == 7 || rate != 2) {
```

Comment:
nope, we're checking here whether we're in an invalid mode. the only valid mode is k=7 & rate=2, so the comparison was right.